conanfile.py

```diff
@@ -11,13 +11,25 @@ class JavaInstallerConan(ConanFile):
     license = "https://www.azul.com/products/zulu-and-zulu-enterprise/zulu-terms-of-use/"
     settings = "os", "arch"
 
-    def config_options(self):    
+    @property
+    def jni_folder(self):
+        if os_info.is_windows:
+            folder = "win32"
+        elif os_info.is_macos:
+            folder = "darwin"
+        elif os_info.is_linux:
+            folder = "linux"
+        else:
+            raise Exception("Unsupported System. This package currently only support Linux/Darwin/Windows")
```

Comment:
Do you mind moving the exception to `config_options()` method, and making it a ConanException? 
`from conans.errors import ConanException`

Reply:
Actually, scratch that.  Lets' restrict the OS and arch using this method instead.  I'm not sure if there are any downsides to this method, but I've seen it in a few high-quality recipes lately and I think it's a good way. 
    settings = {"os" : ["Windows", "Macos", "Linux"], "arch" : ["x86_64"]}

Reply:
That's a good idea, but isn't there a problem when cross-compiling?

If `os == "Android"` I will get an error, right?

Reply:
If that’s an downside of the approach, then just move it to configure. I haven’t had to do crosscompiling. Glad u were here to point it out.

Reply:
Thanks, I've used `tools.detected_architecture` to avoid the same issue.

Just tried with a Android cross-building profile, it works.